packages/clay-badge/src/ClayBadge.soy

```diff
@@ -0,0 +1,110 @@
+{namespace ClayBadge}
+
+/**
+ * This renders a badge.
+ */
+{template .render}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? id: string}
+	{@param? label: html|string}
+	{@param? href: string}
+	{@param? shape: string}
+	{@param? size: string}
+	{@param? style: string}
+
+	{let $attributes kind="attributes"}
+		class="badge
+			{sp}badge-{$style ?: 'default'}
+
+			{if $shape == 'pill'}
+				{sp}badge-pill
+			{/if}
+
+			{if $size}
+				{sp}badge-{$size}
+			{/if}
+		"
+
+		{if $id}
+			id="{$id}"
+		{/if}
+
+		{if $href}
+			href="{$href}"
+		{/if}
+	{/let}
+
+	{if $href}
+		<a {$attributes}>
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
+		</a>
+	{else}
+		<span {$attributes}>
+			{call .content}
+				{param icon: $icon /}
+				{param label: $label /}
+			{/call}
+		</span>
+	{/if}
+
+{/template}
+
+/**
+ * Renders icon content
+ */
+{template .content}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? label: html|string}
+
+	{let $iconAlignment: $icon and $icon.alignment ? $icon.alignment : 'right' /}
+
+	{if $icon and $iconAlignment == 'left'}
```

Comment:
As @carloslancha is hinting, we've been asked several times by UX to keep the components and their variations as limited and restricted as possible. If it isn't designed or approved by the Lexicon team, there's no point in providing the flexibility in the components.

I agree that it's better to start small, with way simpler APIs and then grow from there.